streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -284,14 +295,19 @@ public void testInitializesAndDestroysMetricsReporters() {
         final int oldInitCount = MockMetricsReporter.INIT_COUNT.get();
         final StreamsBuilder builder = new StreamsBuilder();
         final KafkaStreams streams = new KafkaStreams(builder.build(), props);
-        final int newInitCount = MockMetricsReporter.INIT_COUNT.get();
-        final int initDiff = newInitCount - oldInitCount;
-        assertTrue("some reporters should be initialized by calling on construction", initDiff > 0);
 
-        streams.start();
-        final int oldCloseCount = MockMetricsReporter.CLOSE_COUNT.get();
-        streams.close();
-        assertEquals(oldCloseCount + initDiff, MockMetricsReporter.CLOSE_COUNT.get());
+        try {
+            final int newInitCount = MockMetricsReporter.INIT_COUNT.get();
+            final int initDiff = newInitCount - oldInitCount;
+            assertTrue("some reporters should be initialized by calling on construction", initDiff > 0);
+
+            streams.start();
```

Comment:
Ditto here, for this test we can still reuse the shared streams.

Reply:
Not here. We need to get `oldInitCount = MockMetricsReporter.INIT_COUNT.get();` before we create `new KafkaStreams()` instance.